backend/src/test/kotlin/org/loculus/backend/controller/submission/SubmitProcessedDataEndpointTest.kt

```diff
@@ -693,6 +693,71 @@ class SubmitProcessedDataEndpointTest(
         assertThat(response.body(), `is`("FileV2"))
     }
 
+    @Test
+    fun `WHEN the preprocessing pipeline is updated after V2 is submitted THEN new file is public`() {
+        val groupId = groupManagementClient
+            .createNewGroup(group = DEFAULT_GROUP, jwt = jwtForDefaultUser)
+            .andGetGroupId()
+
+        // Step 1: submit entry without files
+        val accession = prepareUnprocessedSequenceEntry(DEFAULT_ORGANISM, groupId = groupId)
+
+        // Step 2: extract and submit processed entry with files
+        val fileIdAndUrlV1 = filesClient.requestUploads(
+            groupId = groupId,
+            jwt = jwtForDefaultUser,
+        ).andGetFileIdsAndUrls()[0]
+        convenienceClient.uploadFile(fileIdAndUrlV1.presignedWriteUrl, "FileV1")
+        convenienceClient.extractUnprocessedData(pipelineVersion = 1)
+
+        submissionControllerClient.submitProcessedData(
+            PreparedProcessedData.withFiles(
+                accession,
+                mapOf("myFileCategory" to listOf(FileIdAndName(fileIdAndUrlV1.fileId, "v1.txt"))),
+            ),
+            pipelineVersion = 1,
+        ).andExpect(status().isNoContent)
+
+        // Step 3: extract and submit processed entry with higher pipeline version and new file
+        val fileIdAndUrlV2 = filesClient.requestUploads(
+            groupId = groupId,
+            jwt = jwtForDefaultUser,
+        ).andGetFileIdsAndUrls()[0]
+        convenienceClient.uploadFile(fileIdAndUrlV2.presignedWriteUrl, "FileV2")
+        convenienceClient.extractUnprocessedData(pipelineVersion = 2)
+
+        submissionControllerClient.submitProcessedData(
+            PreparedProcessedData.withFiles(
+                accession,
+                mapOf("myFileCategory" to listOf(FileIdAndName(fileIdAndUrlV2.fileId, "v2.txt"))),
+            ),
+            pipelineVersion = 2,
+        ).andExpect(status().isNoContent)
+
+        // Step 4: approve entry
```

Comment:
I think it would be good to add a test for what happens when step 3 and step 4 are inverted (maybe you have already covered this)

Reply:
I think that's the test right above

Reply:
yep it is. We have that one covered, that was the issue we had before.

Reply:
Awesome thanks!